pkg/chunked/filesystem_linux.go

```diff
@@ -49,10 +49,31 @@ type fileMetadata struct {
 	skipSetAttrs bool
 }
 
+// splitPath takes a file path as input and returns two components: dir and base.
+// Differently than filepath.Split(), this function handles some edge cases.
+// If the path refers to a file in the root directory, the returned dir is "/".
+// The returned base value is never empty, it never contains any slash and the
+// value "..".
+func splitPath(path string) (string, string, error) {
```

Comment:
In Rust the first would be `Option<str>` or so (e.g. ref [Path::parent](https://doc.rust-lang.org/std/path/struct.Path.html#method.parent)) which would have then forced all call sites to clearly handle its absence or presence in a single canonical form.

Reply:
What do you propose as a Go alternative?

AFAICS the primary value of optionals is that the _compiler_ understands and enforces value presence guarantees on various code paths. We just don’t have that. (The closes option is a pointer type, but Go doesn’t have a “non-`nil` pointer” type in the type system, so, again, no control flow enforcement.)

Reply:
It was one of my usual “aside” comments, not actionable in this PR. I will try to prefix with “Aside:” or so more consistently.

Reply:
No worries — I’m a similarly vocal Go hater, it’s just, as you say, frequently not actionable.